src/bar/bar.test.js

```diff
@@ -0,0 +1,122 @@
+    expect(xAxis).toHaveValue("X")
+    expect(yAxis).toHaveValue("Y")
+})
+
+describe("alerts displayed for missing chart data", function() {
+    test("no data", async function() {
+        initDomFromFiles(html, js)
+        const xAxis = await domTesting.getByLabelText(document, "X label")
+        const yAxis = await domTesting.getByLabelText(document, "Y label")
+        const generateChartBtn = await domTesting.findByText(document, "Generate chart")
+        const spy = jest.spyOn(window, "alert")
+    
+        await userEvent.type(xAxis, "X")
+        await userEvent.type(yAxis, "Y")
+        await userEvent.click(generateChartBtn)
+        
+        expect(spy).toHaveBeenCalledTimes(1)
+        const alert = spy.mock.lastCall[0]
+        expect(alert).toBe("Error: No data specified!")
+        spy.mockRestore()
+    })
+
+    test("no x or y labels", async function() {
+        initDomFromFiles(html, js)
+        const xValue = await domTesting.getByLabelText(document, "X")
+        const yValue = await domTesting.getByLabelText(document, "Y")
+        const generateChartBtn = await domTesting.findByText(document, "Generate chart")
+        const spy = jest.spyOn(window, "alert")
+        
+        await userEvent.type(xValue, "1")
+        await userEvent.type(yValue, "2")
+        await userEvent.click(generateChartBtn)
+        
+        expect(spy).toHaveBeenCalledTimes(1)
+        const alert = spy.mock.lastCall[0]
+        expect(alert).toBe("Error: Must specify a label for both X and Y!")
+        spy.mockRestore()
+    })
+})
+
+test("clearing chart data", async function() {
+    initDomFromFiles(html, js)
+    const addBtn = await domTesting.findByText(document, "+")
+    const clearBtn = await domTesting.findByText(document, "Clear chart data")
+    const title = await domTesting.getByLabelText(document, "Chart title")
+    const color = await domTesting.getByLabelText(document, "Chart color")
+    const xAxis = await domTesting.getByLabelText(document, "X label")
+    const yAxis = await domTesting.getByLabelText(document, "Y label")
+    await userEvent.click(addBtn)
+    xValues = await domTesting.queryAllByLabelText(document, "X")
+    yValues = await domTesting.queryAllByLabelText(document, "Y")
+
+    await userEvent.type(title, "Title")
+
+    color.value = "ffffff"  //don't know if this is allowed but not sure if there's a better way to do it
+                            //feel free to suggest ideas
+    
+    await userEvent.type(xAxis, "X")
+    await userEvent.type(yAxis, "Y")
+    await userEvent.type(xValues[0], "1")
+    await userEvent.type(yValues[0], "2")
+    await userEvent.type(xValues[1], "3")
+    await userEvent.type(yValues[1], "4")
+    await userEvent.click(clearBtn)
+    
+    xValues = await domTesting.queryAllByLabelText(document, "X")
+    yValues = await domTesting.queryAllByLabelText(document, "Y")
+    expect(title).toHaveValue("")
+    expect(color).toHaveValue("#ff4500")
```

Comment:
This does not seem to have been affected by line 101: `color.value = "ffffff"`

Reply:
Because we cleared the chart. Should I assert before that it did get changed?

Reply:
My bad. You are correct it is probably fine since looking back at the project guidelines we don't need to these the changing of color except that it gets reset by clearing